pallets/remarks/src/tests.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+use frame_support::{assert_noop, assert_ok, pallet_prelude::ConstU32, BoundedVec};
+use frame_system::Call as SystemCall;
+use pallet_utility::Call as UtilityCall;
+use sp_runtime::DispatchError::BadOrigin;
+
+use crate::{mock::*, *};
+
+mod remark {
+	use super::*;
+
+	fn get_test_remarks() -> BoundedVec<TestRemark, MaxRemarksPerCall> {
+		BoundedVec::<TestRemark, MaxRemarksPerCall>::try_from(vec![
+			TestRemark::SomeId(1),
+			TestRemark::SomeId(2),
+			TestRemark::SomeBytes(
+				BoundedVec::<u8, ConstU32<16>>::try_from(vec![0, 1, 2])
+					.expect("can build remark bytes"),
+			),
+		])
+		.expect("can build remarks")
+	}
+
+	#[test]
+	fn success() {
+		new_test_ext().execute_with(|| {
+			let origin = RuntimeOrigin::signed(1);
+			let remarks = get_test_remarks();
+
+			let call = RuntimeCall::System(SystemCall::remark {
+				remark: vec![3, 4, 5],
+			});
+
+			let expected_pre_origin = origin.clone();
+			let expected_pre_remarks = remarks.clone();
+			let expected_pre_call = call.clone();
+
+			RemarkDispatchHandlerMock::mock_pre_dispatch_check(move |t| {
```

Comment:
I've never thought of using `mock-builder` to mock the runtime instead of another pallet, but I like it, and it makes a lot of sense! 

Good application 👌🏻